### ISSUE TITLE
Unable to install the package using --extra-index-url

### ISSUE THREAD

**Issue by jmrr** [novemberfiveco/s3pypi#35]
Hi, this follows some comments in #34.

Essentially, I'm trying to install the uploaded package using:

```
pip install --upgrade {mypackage} --extra-index-url https://pypi-index-bucket.com
```
But get timeout responses:

```
Retrying (Retry(total=4, connect=None, read=None, redirect=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7fee0c644710>, 'Connection to pyp-index-bucket.com timed out. (connect timeout=15)')': /mypackage/
```

However, if I link the `index.html` directly:

```
pip install predict --find-links https://s3.eu-west-2.amazonaws.com/pypi-index-bucket.com/mypackage/index.html mypackage
```

I can install the package without issues. But this then would defeat the purpose of `s3pypi`...

Any hint of what can be going wrong? I've followed the steps and at the end, placed the CNAMERecord output in Route53 as an A record.

Thanks!

**Comment by jmrr** [comment]
UPDATE: issue resolved when creating the CloudFormation stack in `us-east-1` rather than `eu-west-2`.

**Comment by jackton1** [comment]
This should be either an issue with the DNS server configuration ensure the that the CloudFront alternate domain is correct and you can resolve the host name `https://pypi-index-bucket.com` to your CloudFront distribution.

**Comment by pabloajz** [comment]
I have exactly the same problem
My Route53 domain resolves properly to the Cloudfront distribution, so it is not that the reason
Any updates on this?

**Comment by derek-adair** [comment]
This was fixed for me by creating a CNAME record pointing to the cloudfront.

**Comment by jgrigd** [comment]
had same problem, worked if I did an A record aliased to cloudfront dist.

**Comment by jbasko** [comment]
I don't understand what does this have to do with regions or CNAME/A records. Isn't this because:

> CloudFront does allow you to specify a default root object (index.html), but it only works on the root of the website (such as http://www.example.com > http://www.example.com/index.html). It does not work on any subdirectory (such as http://www.example.com/about/). If you were to attempt to request this URL through CloudFront, CloudFront would do a S3 GetObject API call against a key that does not exist.

Looks like I'll have to deploy a JavaScript lambda:
https://aws.amazon.com/blogs/compute/implementing-default-directory-indexes-in-amazon-s3-backed-amazon-cloudfront-origins-using-lambdaedge/

**Comment by kutenai** [comment]
I applied the solution that @jbasko described and it worked. I'm not sure why that is not listed anywhere else, and why it isn't part of the 'standard' install of s3pypi.

One issue I have is that s3pipy provides a CloudFront setup only. I'd really prefer some "docs" that describe how to set this up manually since I don't use CloudFront. It seems that this tool might be a tad out of date?

**Comment by sad-boy-uoa** [comment]
Just FYI, there must have been something changed with AWS, but the PEP503 spec requires each repository to return a valid HTML5 page, which has the following structure:

```html
<!DOCTYPE html>
<html>
  <body>
    <a href="/frob/">frob</a>
    <a href="/spamspamspam/">spamspamspam</a>
  </body>
</html>
```

It says that each repo MUST end with a `/` and S3 does not support "default index pages" for subdirectories and so, that edge lambda is required.

I'm not sure the best way to work around this from the POV of this repo other than deploying the edge lambda.

**Comment by mdwint** [comment]
This is fixed in s3pypi 1.0.0, which creates bare index objects by default. See [CHANGELOG.md](https://github.com/novemberfiveco/s3pypi/blob/master/CHANGELOG.md) for details.
